include/deal.II/matrix_free/mapping_info.h

```diff
@@ -446,7 +447,7 @@ namespace internal
       /**
        * The pointer to the underlying Mapping object.
        */
-      SmartPointer<const Mapping<dim>> mapping;
+      std::shared_ptr<dealii::hp::MappingCollection<dim>> mapping;
```

Comment:
There is one minor issue with it - but unrelated to whether we use a `shared_ptr` here or a `unique_ptr` that we would get by cloning: This breaks backward compatibility. While this is an internal class, this is still the way user codes would access the mapping, so I would like to be cautious here. Maybe call the new member `mapping_collection`, to be specific with the name, and initialize the `mapping` as a `SmartPointer` from that? It is a bit redundant but it would be nice if user codes do not break.

Reply:
You mean the smart pointer points to the first entry of the collection? I can do that.

Reply:
Yes exactly, so whenever we get a new shared pointer of mapping collections in a initialize/update_mapping function of this class, we update the smart pointer.

Reply:
Done!